vendor/engines/sanger_sequencing/app/models/sanger_sequencing/ability.rb

```diff
@@ -4,8 +4,14 @@ class Ability
 
     include CanCan::Ability
 
-    def initialize(user)
+    def initialize(user, facility = nil)
+      return unless user
+
       can [:show, :create, :update, :fetch_ids], Submission, user: user
+
+      if facility && user.operator_of?(facility)
```

Comment:
That's a good point. I'm not sure if the ability is the right place to do that. Part of that thought is if it's not enabled, I feel 404 is more appropriate than 403. I'm thinking of doing it in a `before_action`. I'll add that in another PR.